syntax = "proto3";

package TwoProcessConnector;

message InitializeResponse {
  string status = 1;
}

message InitializeRequest {
  string filename = 1;
}

message GetDataResponse {
  string data = 1;
}

message GetDataRequest {
  string req = 1;
}

message ShutdownResponse {
  string status = 1;
}

message ShutdownRequest {
  string options = 1;
}

service Reader {
  rpc initialize(InitializeRequest) returns (InitializeResponse) {}
  rpc getData(GetDataRequest) returns (stream GetDataResponse) {}
  rpc shutdown(ShutdownRequest) returns (ShutdownResponse);
}
